src/libraries/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -402,6 +402,9 @@ public sealed partial class JsonSerializerOptions
         public System.Text.Json.Serialization.JsonConverter GetConverter(System.Type typeToConvert) { throw null; }
         public System.Text.Json.Serialization.Metadata.JsonTypeInfo GetTypeInfo(System.Type type) { throw null; }
         public void MakeReadOnly() { }
+        [System.Diagnostics.CodeAnalysis.RequiresDynamicCodeAttribute("Populating unconfigured TypeInfoResolver properties with the reflection resolver requires runtime code generation.")]
+        [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute("Populating unconfigured TypeInfoResolver properties with the reflection resolver requires unreferenced code.")]
```

Comment:
Should we augment the message to say if you're passing false you should use the other overload instead?

Reply:
I thought about it, but it felt like cramming two separate concerns in the same warning. Most uses would be passing `true` and the warnings are still relevant there.